core/src/main/java/org/springframework/security/access/hierarchicalroles/RoleHierarchyImpl.java

```diff
@@ -282,7 +282,18 @@ private Builder(String rolePrefix) {
 		 */
 		public ImpliedRoles role(String role) {
 			Assert.hasText(role, "role must not be empty");
-			return new ImpliedRoles(role);
+			return new ImpliedRoles(this.rolePrefix.concat(role));
+		}
+
+		/**
+		 * Creates a new hierarchy branch to define an authority and its child roles.
+		 * @param authority the highest authority in this branch
+		 * @return a {@link ImpliedRoles} to define the child roles for the
+		 * <code>authority</code>
+		 */
```

Comment:
Can you please add `@since 6.4`? It is also important to mention that this method does not include the `rolePrefix`.